nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/GetS3ObjectMetadata.java

```diff
@@ -0,0 +1,192 @@
+            .defaultValue(TARGET_ATTRIBUTE)
+            .dependsOn(MODE, MODE_FETCH_METADATA)
+            .build();
+
+    public static final PropertyDescriptor METADATA_ATTRIBUTE = new PropertyDescriptor.Builder()
+            .name("Metadata Attribute")
+            .description("The attribute where the metadata will be written")
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .defaultValue("s3.object.metadata")
+            .dependsOn(METADATA_TARGET, TARGET_ATTRIBUTE)
+            .required(true)
+            .build();
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    public static final List<PropertyDescriptor> properties = List.of(
+            MODE,
+            METADATA_TARGET,
+            METADATA_ATTRIBUTE,
+            BUCKET_WITH_DEFAULT_VALUE,
+            KEY,
+            AWS_CREDENTIALS_PROVIDER_SERVICE,
+            S3_REGION,
+            TIMEOUT,
+            FULL_CONTROL_USER_LIST,
+            READ_USER_LIST,
+            READ_ACL_LIST,
+            OWNER,
+            SSL_CONTEXT_SERVICE,
+            ENDPOINT_OVERRIDE,
+            SIGNER_OVERRIDE,
+            S3_CUSTOM_SIGNER_CLASS_NAME,
+            S3_CUSTOM_SIGNER_MODULE_LOCATION,
+            PROXY_CONFIGURATION_SERVICE);
+
+    public static Relationship REL_FOUND = new Relationship.Builder()
+            .name("found")
+            .description("An object was found in the bucket at the supplied key")
+            .build();
+
+    public static Relationship REL_NOT_FOUND = new Relationship.Builder()
+            .name("not found")
+            .description("No object was found in the bucket the supplied key")
+            .build();
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return new HashSet<>(List.of(REL_FOUND, REL_NOT_FOUND, REL_FAILURE));
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        boolean isRouter = context.getProperty(MODE).getValue().equals(MODE_ROUTER.getValue());
+
+        final AmazonS3Client s3;
+        try {
+            s3 = getS3Client(context, flowFile.getAttributes());
+        } catch (Exception e) {
+            getLogger().error("Failed to initialize S3 client", e);
+            flowFile = session.penalize(flowFile);
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final String bucket = context.getProperty(BUCKET_WITH_DEFAULT_VALUE).evaluateAttributeExpressions(flowFile).getValue();
+        final String key = context.getProperty(KEY).evaluateAttributeExpressions(flowFile).getValue();
+
+        try {
+            Relationship route;
```

Comment:
This should be named `relationship`.
```suggestion
            Relationship relationship;
```